src/com/oneofever/functions/History.java

```diff
@@ -0,0 +1,24 @@
+/* (C)2024 - one-of-the-teams-ever */
+package com.oneofever.functions;
+
+import com.oneofever.Pair;
+import com.oneofever.shapes.Shape;
+import java.util.ArrayList;
+import java.util.Comparator;
+
+public class History {
+    static ArrayList<Pair<String, Shape>> createdShapes = new ArrayList<Pair<String, Shape>>();
+
+    private History() {}
+
+    public static void add(String name, Shape toAdd) {
+        createdShapes.add(new Pair<>(name, toAdd));
+    }
+
+    public static String[] getHistory() {
```

Comment:
returning `Shape[]` might've been better, idk

Reply:
It can be changed in the future pretty easily if we need it